share/completions/p4.fish

```diff
@@ -0,0 +1,803 @@
+# Fish completions for Perforce (p4) SCM
+# Based on the list of commands from
+# https://www.perforce.com/perforce/r16.1/manuals/cmdref/
+# Author: Boris Aranovich https://github.com/nomaed
+
+#########################################################
+### p4 command parsing
+#########################################################
+
+function __fish_print_p4_client_name
+    set -l p4info (p4 info -s 2> /dev/null)
+    if test -z "$p4info"
```

Comment:
True, to avoid errors. However, there is a `string match` and a `string match | string replace` later on that will be executed. The test eliminates these 3 commands in case there's no `p4 info` result. That's my consideration.

Although I guess that the typical use case for this completion would be some user that has p4 set up and is in a p4 workspace. Otherwise the completions won't even be used anyhow. So it's probably safe to assume that this test will not be a benefit in real use case.

Can you estimate whether having this test would be non-negligible performance-wise?